### PR TITLE
Convert HTML to markdown for 2.2 and 2.1 on the database caching page

### PR DESCRIPTION
**Pull request by leviv**
## This PR is a:

- [ ] New topic
- [ ] Content update
- [x] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary
Convert HTML to markdown for 2.2 and 2.1 on the database caching page

When this pull request is merged, it will...
Make the source code easier to read

Replaces some HTML tags with markdown equivalent

### CODE REVIEW

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 1]
```diff
@@ -13,29 +13,30 @@ redirect_from:
   -  /guides/v2.2/config-guide/database/database.html
 ---
 
-This topic discusses how to use the Magento 2 database for caching. After you complete these tasks, cached objects are stored in the `cache` and `cache_tag` Magento 2 database tables. Nothing is stored in `var/cache` or `var/page_cache`.
+## Overview of database caching {#mage-cache-db-over}
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: This heading is unnecessary. I know that there are several other topics on devdocs that use an introductory heading like this, but it's not necessary.

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 2]
```diff
@@ -13,29 +13,30 @@ redirect_from:
...
+This topic discusses how to use the Magento 2 database for caching. After you complete these tasks, cached objects are stored in the `cache` and `cache_tag` Magento 2 database tables. Nothing is stored `var/cache` or `var/page_cache`.
 
 This topic discusses how to set up database caching and how to verify database caching is working. We discuss the following options:
 
 *	Using the `default` cache frontend, in which case you modify `di.xml` only.
 *	Using a custom {% glossarytooltip 0bc9c8bc-de1a-4a06-9c99-a89a29c30645 %}cache{% endglossarytooltip %} frontend, in which case you modify `env.php` only.
 
 <div class="bs-callout bs-callout-warning">
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: For notes, please use the following syntax:

```
{:.bs-callout .bs-callout-info}
This is a note.
```

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 3]
```diff
@@ -13,29 +13,30 @@ redirect_from:
...
-    <p>Database caching&mdash;like file-based caching&mdash; works well in a development environment but we <em>strongly recommend</em> you use <a href="{{ page.baseurl }}/config-guide/varnish/config-varnish.html">Varnish</a> in production instead.</p>
-    <p>Varnish is designed to accelerate the HTTP protocol.</p>
+Database caching&mdash;like file-based caching&mdash; works well in a development environment but we <em>strongly recommend</em> you use [Varnish] in production instead.
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Replace HTML emphasis tags with underscores or single asterisks. 

```
_strongly recommended_
```

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 4]
```diff
@@ -13,29 +13,30 @@ redirect_from:
...
+Varnish is designed to accelerate the HTTP protocol.
 </div>
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Remove closing div tag

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 5]
```diff
@@ -72,11 +73,11 @@ To modify `di.xml`:
         	<arguments>
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Wrap code blocks in code fences (```) and specify the code language. For example:

```
```xml
<type name="Magento\Framework\App\Cache\Frontend\Pool">
   <arguments>
      <argument name="frontendSettings" xsi:type="array">
          <item name="page_cache" xsi:type="array">
              <item name="backend_options" xsi:type="array">
                <item name="cache_dir" xsi:type="string">page_cache</item>
              </item>
          </item>
      </argument>
   </arguments>
</type>
<type name="Magento\Framework\App\Cache\Type\FrontendPool">
   <arguments>
      <argument name="typeFrontendMap" xsi:type="array">
        <item name="full_page" xsi:type="string">page_cache</item>
      </argument>
   </arguments>
</type>
```
```

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 6]
```diff
@@ -130,77 +131,78 @@ To enable database caching using a custom cache frontend, you must modify `<your
...
 To verify database caching is working, clear the current cache directories, go to any cacheable page in a web browser, and verify that data is written to the database and not to the file system.
 
 Use the following steps:
 
-1.	If you haven't done so already, log in to the Magento server as, or switch to, the <a href="{{ page.baseurl }}/install-gde/prereq/file-sys-perms-over.html">Magento file system owner</a>.
+1.	If you haven't done so already, log in to the Magento server as, or switch to, the [Magento file system owner].
 2.	Clear the current cache directories:
 
-		rm -rf <your Magento install dir>/var/cache/* <your Magento install dir>/var/page_cache/* <your Magento install dir>/var/di/* <your Magento install dir>/var/generation/*
+		rm -rf <your Magento install dir>/var/cache/* <your Magento install dir>/var/page_cache/* <your Magento install dir>/generated/metadata/* <your Magento install dir>/generated/code/*
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Wrap command in code fences (```) and specify that it's bash. For example:

```

```bash
rm -rf <your Magento install dir>/var/cache/* <your Magento install dir>/var/page_cache/* <your Magento install dir>/generated/metadata/* <your Magento install dir>/generated/code/*
```

```

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 7]
```diff
@@ -130,77 +131,78 @@ To enable database caching using a custom cache frontend, you must modify `<your
...
 3.	In a web browser, go to any cacheable page (such as the {% glossarytooltip 1a70d3ac-6bd9-475a-8937-5f80ca785c14 %}storefront{% endglossarytooltip %} front door page).
 
-	If exceptions display, verify `di.xml` syntax and try again. (To see exceptions in the browser, you must <a href="{{ page.baseurl }}/config-guide/cli/config-cli-subcommands-mode.html#config-mode">enable developer mode</a>.)
+	If exceptions display, verify `di.xml` syntax and try again. (To see exceptions in the browser, you must [enable developer mode].)
 4.	Enter the following commands:
 
 		ls <your Magento install dir>/var/cache/*
 		ls <your Magento install dir>/var/page_cache/*
 
     <div class="bs-callout bs-callout-info" id="info">
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Convert HTML note syntax to markdown:

```
{:.bs-callout .bs-callout-info}
This is a note}
```

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 8]
```diff
@@ -1,11 +1,7 @@
 ---
 group: extension-dev-guide
-subgroup: 08_Partial caching
 title: Create custom cache engines
-menu_title: Create custom cache engines
-menu_order: 9
-menu_node:
-version: 2.2
+version: 2.3
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: This should be version 2.0

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 9]
```diff
@@ -136,52 +132,53 @@ To enable database caching using a custom cache frontend, you must modify `<your
...
 3.	In a web browser, go to any cacheable page (such as the {% glossarytooltip 1a70d3ac-6bd9-475a-8937-5f80ca785c14 %}storefront{% endglossarytooltip %} front door page).
 
-	If exceptions display, verify `di.xml` syntax and try again. (To see exceptions in the browser, you must <a href="{{ page.baseurl }}/config-guide/cli/config-cli-subcommands-mode.html#change-to-developer-mode">enable developer mode</a>.)
+	If exceptions display, verify `di.xml` syntax and try again. (To see exceptions in the browser, you must [enable developer mode].)
 4.	Enter the following commands:
 
 		ls <your Magento install dir>/var/cache/*
 		ls <your Magento install dir>/var/page_cache/*
 
     <div class="bs-callout bs-callout-info" id="info">
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Use markdown syntax for notes

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 10]
```diff
@@ -136,52 +132,53 @@ To enable database caching using a custom cache frontend, you must modify `<your
...
+## Verify database caching is working {#mage-cache-db-verify}
 To verify database caching is working, clear the current cache directories, go to any cacheable page in a web browser, and verify that data is written to the database and not to the file system.
 
 Use the following steps:
 
-1.	If you haven't done so already, log in to the Magento server as, or switch to, the <a href="{{ page.baseurl }}/install-gde/prereq/file-sys-perms-over.html">Magento file system owner</a>.
+1.	If you haven't done so already, log in to the Magento server as, or switch to, the [Magento file system owner].
 2.	Clear the current cache directories:
 
 		rm -rf <your Magento install dir>/var/cache/* <your Magento install dir>/var/page_cache/* <your Magento install dir>/generated/metadata/* <your Magento install dir>/generated/code/*
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Wrap command in code fences

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 11]
```diff
@@ -98,19 +94,19 @@ To modify `di.xml`:
...
 5.	Save your changes to `di.xml` and exit the text editor.
 
-7.	Continue with <a href="#mage-cache-db-verify">Verify database caching is working</a>.
+7.	Continue with [Verify database caching is working].
 
-<h2 id="mage-cache-db-env">Database caching using a custom cache frontend</h2>
+## Database caching using a custom cache frontend {#mage-cache-db-env}
 This section discusses how to set up database caching with a custom {% glossarytooltip ca5ad9ac-9d39-45b5-80b1-e90d192f20d0 %}cache frontend{% endglossarytooltip %}.
 
 <div class="bs-callout bs-callout-info" id="info">
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Use markdown syntax for notes

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 12]
```diff
@@ -98,19 +94,19 @@ To modify `di.xml`:
...
 <span class="glyphicon-class">
-  <p>Due to a known issue, a custom cache frontend still results in some objects being cached to the file system; however, fewer assets are cached compared to file system caching.</p></span>
+  Due to a known issue, a custom cache frontend still results in some objects being cached to the file system; however, fewer assets are cached compared to file system caching.</span>
 </div>
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Remove div

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 13]
```diff
@@ -27,21 +23,21 @@ This topic discusses how to set up database caching and how to verify database c
 *	Using a custom {% glossarytooltip 0bc9c8bc-de1a-4a06-9c99-a89a29c30645 %}cache{% endglossarytooltip %} frontend, in which case you modify `env.php` only.
 
 <div class="bs-callout bs-callout-warning">
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Use markdown syntax for notes

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 14]
```diff
@@ -27,21 +23,21 @@ This topic discusses how to set up database caching and how to verify database c
...
-    <p>Database caching&mdash;like file-based caching&mdash; works well in a development environment but we <em>strongly recommend</em> you use <a href="{{ page.baseurl }}/config-guide/varnish/config-varnish.html">Varnish</a> in production instead.</p>
-    <p>Varnish is designed to accelerate the HTTP protocol.</p>
+Database caching&mdash;like file-based caching&mdash; works well in a development environment but we <em>strongly recommend</em> you use [Varnish] in production instead.
+Varnish is designed to accelerate the HTTP protocol.
 </div>
```
Comment (jeff-matthews) [inline guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Remove div

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 8 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Why not 2.2?

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 5 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 1 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 2 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 3 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 15]
```diff
@@ -87,24 +87,24 @@ To modify `di.xml`:
...
-7.	Continue with <a href="#mage-cache-db-verify">Verify database caching is working</a>.
+7.	Continue with [Verify database caching is working].
 
-<h2 id="mage-cache-db-env">Database caching using a custom cache frontend</h2>
+## Database caching using a custom cache frontend {#mage-cache-db-env}
 This section discusses how to set up database caching with a custom {% glossarytooltip ca5ad9ac-9d39-45b5-80b1-e90d192f20d0 %}cache frontend{% endglossarytooltip %}.
 
-<div class="bs-callout bs-callout-info" id="info">
+{:.bs-callout .bs-callout-info #info}
 <span class="glyphicon-class">
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Remove HTML span element

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md [hunk 16]
```diff
@@ -130,77 +130,80 @@ To enable database caching using a custom cache frontend, you must modify `<your
...
 		ls <your Magento install dir>/var/cache/*
 		ls <your Magento install dir>/var/page_cache/*
 
-    <div class="bs-callout bs-callout-info" id="info">
-      <span class="glyphicon-class">
-      <p>Due to a known issue, a custom cache frontend still results in some objects being cached to the file system; however, fewer assets are cached compared to file system caching.</p>
-      <p>If you use the <code>default</code> cache frontend, you don't have this issue.</p></span>
-    </div>
+    {:.bs-callout .bs-callout-info #info}
+    <span class="glyphicon-class">
```
Comment (jeff-matthews) [inline guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Remove HTML span element

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 4 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 6 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 7 above)
Comment (leviv) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 9 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 10 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 11 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 12 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 13 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

File: guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 14 above)
Comment (leviv) [reply guides/v2.2/extension-dev-guide/cache/partial-caching/database-caching.md]: Fixed

**Comment by leviv** [comment]
@jeff-matthews  Requested changes made

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 15 above)
Comment (jeff-matthews) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: @leviv, I think I made this comment as you were making changes to the callout classes. Please remove span.

File: guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md (hunk 16 above)
Comment (jeff-matthews) [reply guides/v2.1/extension-dev-guide/cache/partial-caching/database-caching.md]: @leviv, I think I made this comment as you were making changes to the callout classes. Please remove span.

**Comment by leviv** [comment]
@jeff-matthews removed span

**Pull request merged**